qrexec/tests/socket/agent.py

```diff
@@ -148,28 +151,52 @@ def test_handshake(self):
         dom0 = self.connect_dom0()
         dom0.handshake()
 
-    def test_just_exec(self):
+    def _test_just_exec(self, cmd):
+        self.assertEqual(cmd[-1], 0)
         self.start_agent()
 
         dom0 = self.connect_dom0()
         dom0.handshake()
 
         user = getpass.getuser().encode("ascii")
 
-        cmd = ("touch " + os.path.join(self.tempdir, "new_file")).encode(
-            "ascii"
-        )
         dom0.send_message(
             qrexec.MSG_JUST_EXEC,
             struct.pack("<LL", self.target_domain, self.target_port)
             + user
             + b":"
             + cmd
-            + b"\0",
```

Comment:
Why removing it here and forcing every caller to include \0 in the string? Are you expecting some (future?) call to not have this final NUL byte?

Reply:
I think it made some assertions easier, but reverted.